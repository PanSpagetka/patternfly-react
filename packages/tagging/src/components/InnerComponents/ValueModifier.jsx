import React from 'react';
import PropTypes from 'prop-types';
import { Col, ControlLabel, FormGroup } from 'patternfly-react';
import ValueSelector from './ValueSelector';

const ValueModifier = ({
  tagValues, onTagValueChange, selectedTagValue, multiValue, valueLabel,
}) => (
  <FormGroup>
    <Col xs={12} md={4} lg={6}>
      <ControlLabel>{valueLabel}</ControlLabel>
    </Col>
    <Col xs={12} md={8} lg={6}>
      <ValueSelector tagValues={tagValues} onTagValueChange={onTagValueChange} selectedOption={selectedTagValue} multiValue={multiValue} />
    </Col>
  </FormGroup>
);

ValueModifier.propTypes = {
  selectedTagValue:PropTypes.shape({ id: PropTypes.number, description: PropTypes.string }),
  onTagValueChange: PropTypes.func.isRequired,
  valueLabel: PropTypes.string,
  multiValue: PropTypes.bool,
  tagValues:PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.number.isRequired, description: PropTypes.string.isRequired }).isRequired).isRequired,
};

ValueModifier.defaultProps = {
  valueLabel: 'Value',
  multiValue: true,
  selectedOption: {},
};

export default ValueModifier;
