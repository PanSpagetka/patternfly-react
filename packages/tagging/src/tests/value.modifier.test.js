import React from 'react';
import ValueModifier from '../components/ValueModifier';
import renderer from 'react-test-renderer';

const selectedTagValue = { description: 'Duck', id: 1 };
const onTagValueChange = jest.fn();
const tagValues = [
  { description: 'Duck', id: 1 },
  { description: 'Cat', id: 2 },
  { description: 'Dog', id: 3 }
];

describe('TagCategory Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(
      <ValueModifier
        onTagValueChange={onTagValueChange}
        selectedTagValue={selectedTagValue}
        multiValue={false}
        tagValues={tagValues}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
