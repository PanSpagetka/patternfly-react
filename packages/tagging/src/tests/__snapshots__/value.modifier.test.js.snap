// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCategory Component match snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="col-lg-6 col-md-4 col-xs-12"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      Value
    </label>
  </div>
  <div
    className="col-lg-6 col-md-8 col-xs-12"
  >
    <div
      className="Select has-value is-searchable Select--single"
      style={undefined}
    >
      <input
        disabled={false}
        name="form-field-name"
        type="hidden"
        value="1"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-2--value-item"
              role="option"
            >
              Duck
            </span>
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
